dsp/src/lockin.rs

```diff
@@ -52,47 +52,36 @@
 //! the demodulation frequency. This does not require any state
 //! information and is therefore a normal function.
 
-use super::iir::{IIRState, IIR};
-use super::Complex;
-use core::f32::consts::PI;
+use super::iir_int::{IIRState, IIR};
+use super::pll::PLL;
+use super::trig::{atan2, cossin};
+use super::{divide_round, Complex};
 
-/// The number of ADC samples in one batch.
-pub const ADC_SAMPLE_BUFFER_SIZE: usize = 16;
-/// The number of outputs sent to the DAC for each ADC batch.
-pub const DECIMATED_BUFFER_SIZE: usize = 1;
+/// TODO these constants are copied from main.rs and should be
```

Comment:
I think if you remove the wrappers and move code into the `process()` ISR in `main.rs` then these will move as well and the TODO is resolved.